backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -21,5 +27,22 @@ public async Task<UserDto> CreateUser(UserRegisterDto userDto)
 
             return _mapper.Map<UserDto>(userEntity);
         }
+
+        public async Task<LessonDto> EnrollUserToLesson(long lessonId)
+        {
+            var user = _context.Users
+                .Single(p => p.Id == _firebaseAuthService.UserId);
+
+            Lesson existingLesson = _context.Lessons
+                .Single(p => p.Id == lessonId);
+
+            user.Lessons.Add(existingLesson);
+
+            await _context.SaveChangesAsync();
+
+            return _mapper.Map<LessonDto>(existingLesson, options => options.AfterMap((o, dto) =>
+                  dto.SubscribersCount = _context.Lessons.Select(t => new { Id = t.Id, SbCount = t.Subscribers.Count })
+                      .Single(l => l.Id == lessonId).SbCount));
+        }
```

Comment:
please create a separate method for this instead of AfterMap

Reply:
still not fixed

Reply:
Created as a separate inline method. Ordinary method can't be used because we need to capture `lessonId`.